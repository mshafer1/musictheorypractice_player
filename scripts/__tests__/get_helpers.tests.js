const get_helpers = require('../get_helpers.js');

describe('_load_get', () => {
    var empty = {};
    it('returns empty for empty query', () => {
        expect(get_helpers._load_get('ahkgen.com/'))
            .toEqual(empty);
    });

    it('takes values into object', () => {
        expect(get_helpers._load_get('ahkgen.com/?length=0'))
            .toEqual({ 'length': "0" });
    });

    it('takes array values into array', () => {
        expect(get_helpers._load_get('ahkgen.com/?skey0%5B%5D=CTRL&skey0%5B%5D=ALT'))
            .toEqual({ 'skey0[]': ['CTRL', 'ALT'] })
    });

    it('takes index array and turns into array', () => {
        expect(get_helpers._load_get('ahkgen.com/?indexes=0'))
            .toEqual({ 'indexes': '0' })
    });

    it('takes url with extra spaces and trims them', () => {
        expect(get_helpers._load_get('ahkgen.com/?   indexes=0'))
            .toEqual({ 'indexes': '0' })
    });

    it('takes url with url escaped spaces and keeps them', () => {
        expect(get_helpers._load_get('ahkgen.com/?+++indexes=0'))
            .toEqual({ '   indexes': '0' })
    });

    it('take a url and does not strip values', () => {
        expect(get_helpers._load_get('ahkgen.com/?+++indexes=+'))
            .toEqual({ '   indexes': ' ' })
    });

    it('takes duplicate values into array', () => {
        expect(get_helpers._load_get('ahkgen.com/?skey0=CTRL&skey0=ALT'))
            .toEqual({ 'skey0': ['CTRL', 'ALT'] })
    });

    it('takes a basic query and returns expected data', () => {
        const result = get_helpers
            ._load_get('ahkgen.com/?length=1&comment0=&func0=KEY&skeyValue0=a&input0=b&option0=Replace');
        expect(result).toMatchSnapshot();
    })

    it('takes data in regardless of values', () => {
        const result = get_helpers
            ._load_get('ahkgen.com/?length=1&comment5=&func5=KEY&skeyValue5=a&input5=b&option5=Replace');
        expect(result).toMatchSnapshot();
    });

    it('takes no issue with an indexes query', () => {
        const result = get_helpers
        ._load_get(
            'ahkgen.com/?indexes=1&comment0=&func0=KEY&skeyValue0=a&input0=b&option0=Send'
        );
        expect(result).toMatchSnapshot();
    })
})
