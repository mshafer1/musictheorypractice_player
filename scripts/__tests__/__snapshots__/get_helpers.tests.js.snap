// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_load_get takes a basic query and returns expected data 1`] = `
Object {
  "comment0": "",
  "func0": "KEY",
  "input0": "b",
  "length": "1",
  "option0": "Replace",
  "skeyValue0": "a",
}
`;

exports[`_load_get takes data in regardless of values 1`] = `
Object {
  "comment5": "",
  "func5": "KEY",
  "input5": "b",
  "length": "1",
  "option5": "Replace",
  "skeyValue5": "a",
}
`;

exports[`_load_get takes no issue with an indexes query 1`] = `
Object {
  "comment0": "",
  "func0": "KEY",
  "indexes": "1",
  "input0": "b",
  "option0": "Send",
  "skeyValue0": "a",
}
`;
